receiver/scraperhelper/scrapercontroller_test.go

```diff
@@ -352,12 +363,83 @@ func TestSingleScrapePerTick(t *testing.T) {
 
 	tickerCh <- time.Now()
 
-	assert.Equal(t, 1, <-scrapeMetricsCh)
+	assert.Eventually(
+		t,
+		func() bool {
+			return <-scrapeMetricsCh == 2
+		},
+		300*time.Millisecond,
+		100*time.Millisecond,
+		"Make sure the scraper channel is called twice",
+	)
 
 	select {
 	case <-scrapeMetricsCh:
-		assert.Fail(t, "Scrape was called more than once")
+		assert.Fail(t, "Scrape was called more than twice")
 	case <-time.After(100 * time.Millisecond):
 		return
 	}
 }
+
+func TestScrapeControllerStartsOnInit(t *testing.T) {
+	t.Parallel()
+
+	tsm := &testScrapeMetrics{
+		ch: make(chan int, 1),
+	}
+
+	scp, err := NewScraper("", tsm.scrape)
+	require.NoError(t, err, "Must not error when creating scraper")
+
+	r, err := NewScraperControllerReceiver(
+		&ScraperControllerSettings{
+			CollectionInterval: time.Hour,
+			InitialDelay:       0,
+		},
+		receivertest.NewNopCreateSettings(),
+		new(consumertest.MetricsSink),
+		AddScraper(scp),
+	)
+	require.NoError(t, err, "Must not error when creating scrape controller")
+
+	assert.NoError(t, r.Start(context.Background(), componenttest.NewNopHost()), "Must not error on start")
+	<-time.After(500 * time.Nanosecond)
+	assert.NoError(t, r.Shutdown(context.Background()), "Must not have errored on shutdown")
+	assert.Equal(t, tsm.timesScrapeCalled, 1, "Must have been called as soon as the controller started")
```

Comment:
Expected value should be first

```suggestion
	assert.Equal(t, 1, tsm.timesScrapeCalled, "Must have been called as soon as the controller started")
```